sshd_config

```diff
@@ -17,3 +17,7 @@ Subsystem sftp	/usr/libexec/openssh/sftp-server
 
 # Configured by user data
 TrustedUserCAKeys /etc/ssh/trusted_user_ca_keys.pub
+
+# EC2 instance connect
+AuthorizedKeysCommand /opt/aws/bin/eic_run_authorized_keys %u %f
```

Comment:
From my perusal of the ssh docs, its seems to me that this will always run.  Am I reading that correctly?

Reply:
https://github.com/aws/aws-ec2-instance-connect-config/blob/master/src/deb_systemd/ssh.service.d/ec2-instance-connect.conf#L3

Yes, this is the main entry point of ec2 instance connect.

https://github.com/aws/aws-ec2-instance-connect-config/blob/master/src/bin/eic_run_authorized_keys
It is a thin wrapper that calls eic_curl_authorized_keys

https://github.com/aws/aws-ec2-instance-connect-config/blob/master/src/bin/eic_curl_authorized_keys
The eic_curl_authorized_keys fetches valid key signature from the ec2 instance connect metadata servers, and validate ssh requests. Therefore, it should be always run to use ec2-instance-connect.